### DISCUSSION TITLE
大佬，路由器openwrt上跑，我感觉结果不准

### DISCUSSION THREAD

**Discussion by wwwliutao125** [XIU2/CloudflareSpeedTest#352] [讨论]
前几天我咨询过，现在后半夜跑完，上午的速度还行，下午的不快，晚上更慢，没有windows上运行的准。ip是换了，但是不知道为啥不准。跑的时候肯定是没挂梯子。是不是脚本有问题

**Comment by wwwliutao125** [comment]
这图windows选的ip测速
![image](https://user-images.githubusercontent.com/20475019/106693647-dccf9180-6611-11eb-9b81-6be569136a96.png)

**Comment by XIU2** [comment]
和脚本无关，脚本只是运行 CloudflareST 并从结果文件中获得最快的 IP。

这种情况我也不是很清楚，下载测速是把文件缓存到内存中并舍弃，难道是因为路由器内存太小导致的？

****

你的意思是，你在路由器上获得最快 IP 后，上午用着速度很快，但是下午就速度下降，晚上就非常慢了？

因为 Cloudflare CDN 用的是 Anycast（任播技术），这意味着**每个 CDN IP 都是动态**的，你测速获得最快 IP 后，可能过一会就调整了，也可能过几天才调整，你说的情况，很可能是正好碰到 CDN 动态调整规律了。

**Comment by wwwliutao125** [comment]
你等，我再测速路由器选的ip。我感觉就是不准。几天上午速度也不是很快，只能说凑合

**Comment by wwwliutao125** [comment]
路由ip1测速，，，![image](https://user-images.githubusercontent.com/20475019/106694124-e3123d80-6612-11eb-8ca0-32bf29c28d13.png)

**Comment by XIU2** [comment]
CloudflareST 也可以单独测速 IP 的。
在 CloudflareST 文件夹下新建文本文件.txt 并重命名为 `1.txt`，写入：`104.17.220.29/32` 并保存。
然后通过 CMD 运行 CloudflareST 时加上 `-f 1.txt` 参数，这样就只会测速这一个 IP 了。

**Comment by wwwliutao125** [comment]
ip2行，![image](https://user-images.githubusercontent.com/20475019/106694375-60d64900-6613-11eb-85d9-dd5c0a63d30e.png)

**Comment by wwwliutao125** [comment]
路由ip3也不行![image](https://user-images.githubusercontent.com/20475019/106694567-ce827500-6613-11eb-91da-46e7ca71d2ca.png)

**Comment by XIU2** [comment]
路由器上面单独测速 IP 同上。
你可以分别在路由器上和电脑上用 CloudflareST 测速同一个 IP。

这样才有参考价值（其他软件和我的软件运行原理不同）。

**Comment by wwwliutao125** [comment]
这是同样在windows用别人的测速ip，有参考价值，

**Comment by XIU2** [comment]
如果你不按我说的做，那。。。

**Comment by wwwliutao125** [comment]
你这个路由器这里出来的ip，感觉有很多是被防火墙阻拦的，不是很通。https://github.com/badafans/better-cloudflare-ip   我用这个测的

**Comment by wwwliutao125** [comment]
关键我不太会

> CloudflareST 也可以单独测速 IP 的。
> 在 CloudflareST 文件夹下新建文本文件.txt 并重命名为 `1.txt`，写入：`104.17.220.29/32` 并保存。
> 然后通过 CMD 运行 CloudflareST 时加上 `-f 1.txt` 参数，这样就只会测速这一个 IP 了。

**Comment by wwwliutao125** [comment]
你等等啊

**Comment by XIU2** [comment]
那就远程协助吧。

**Comment by wwwliutao125** [comment]
好啊

**Comment by XIU2** [comment]
问题解决，如果再出现什么其他问题，可以在这里告诉我。
